akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -372,8 +372,13 @@ private[remote] object EndpointManager {
      */
     def markAsFailed(endpoint: ActorRef, timeOfRelease: Deadline): Unit =
       if (isWritable(endpoint)) {
-        addressToWritable += writableToAddress(endpoint) -> Gated(timeOfRelease)
-        writableToAddress -= endpoint
+        val address = writableToAddress(endpoint)
+        addressToWritable.get(address) match {
+          case Some(Quarantined(_, _)) ⇒ // don't overwrite Quarantined with Gated
+          case _ ⇒
```

Comment:
Perhaps better keep it unchanged, and add keepQuarantedOr logic in the caller's code？